src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.cpp

```diff
@@ -388,6 +393,7 @@ void TlsCertificateManagementServer::HandleProvisionClientCertificate(HandlerCon
         ReturnOnFailure(Crypto::IsCertificateValidAtCurrentTime(cert),
                         ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::DynamicConstraintError));
     }
+    ReturnOnFailure(srcIter.GetStatus(), ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError));
```

Comment:
```suggestion
    ReturnOnFailure(srcIter.GetStatus(), ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidAction));
```

Malformed data is not a constraint violation, it's invalid.